homeassistant/components/wwlln/config_flow.py

```diff
@@ -60,11 +61,16 @@ class WWLLNFlowHandler(config_entries.ConfigFlow):
         else:
             user_input[CONF_UNIT_SYSTEM] = CONF_UNIT_SYSTEM_METRIC
 
-        # To simplify things, we don't allow users of the config flow to
-        # input a window; instead, we make a sane assumption to use the
-        # default (stored as seconds, since timedelta's aren't
-        # JSON-serializable):
-        if CONF_WINDOW not in user_input:
+        # When importing from `configuration.yaml`, we give the user
+        # flexibility by allowing the `window` parameter to be any type
+        # of time period. This will always return a timedelta; unfortunately,
+        # timedeltas aren't JSON-serializable, so we can't store them in a
+        # config entry as-is; instead, we save the total seconds as an int:
+        if CONF_WINDOW in user_input:
+            if isinstance(user_input[CONF_WINDOW], timedelta):
```

Comment:
We shouldn't need this check.

Reply:
The best way to test this, would be to set up the component. Then the config validation would run and we would test what is actually passed.

Reply:
😴